Florida_Analysis_Py_File.py

```diff
@@ -0,0 +1,534 @@
+    "Overdose_Per_100K"
+].mean()
+florida_results_yearwise
+
+florida_result_copy = florida_result.copy()
+
+# create a scale for number of years before and after 2010 (target year)
+def scale_years(year):
+    if year == 2007:
+        return -3
+    if year == 2008:
+        return -2
+    if year == 2009:
+        return -1
+    if year == 2010:
+        return 0
+    if year == 2011:
+        return 1
+    if year == 2012:
+        return 2
+    if year == 2013:
+        return 3
+
+
+florida_result_copy["Year Relative To Policy"] = florida_result_copy["Year"].apply(
+    lambda x: scale_years(x)
+)
+
+# double check no nulls in "year relative to policy"
+
+assert florida_result_copy["Year Relative To Policy"].isnull().sum() == 0
+
+# doing this in case the float == int comparison causes issues
+
+# split into before 2010 and after 2010
+
+florida_mortality_before_policy_change = florida_result_copy[
+    florida_result_copy["Year"] < 2010
+]
+florida_mortality_after_policy_change = florida_result_copy[
+    florida_result_copy["Year"] > 2010
+]
+
+from sklearn.linear_model import LinearRegression
+
+regressor_before = LinearRegression()
+regressor_after = LinearRegression()
+
+
+x_before = np.array(
+    florida_mortality_before_policy_change["Year Relative To Policy"]
+).reshape(-1, 1)
+y_before = np.array(
+    florida_mortality_before_policy_change["Overdose_Per_100K"]
+).reshape(-1, 1)
+
+x_after = np.array(
+    florida_mortality_after_policy_change["Year Relative To Policy"]
+).reshape(-1, 1)
+y_after = np.array(florida_mortality_after_policy_change["Overdose_Per_100K"]).reshape(
+    -1, 1
+)
+
+
+regressor_before.fit(x_before, y_before)
+regressor_after.fit(x_after, y_after)
+
+
+y_pred_before = regressor_before.predict(x_before)
+y_pred_after = regressor_after.predict(x_after)
+
+import matplotlib.pyplot as plt
+
+plt.xlim(-3, 3)
+# plt.ylim(0, 500)
+
+plt.xlabel("Year Relative To Policy Change Year (2010)")
+plt.ylabel("Deaths due to Drug Overdose (Per 100K)")
+plt.title("Pre-Post Analysis Of Drug Overdose Deaths in Florida")
+
+plt.plot(x_before, y_pred_before, color="blue")
+plt.plot(x_after, y_pred_after, color="red")
+
+plt.grid()
+# plot avg value in each year
+
+## Difference-In-Difference Analysis Of Drug Overdose Deaths In Florida
+
+# creating a new dataset, that contains drug related deaths in other neighboring states of Florida - Alabama, Georgia, Mississippi, South Carolin and Tennessee
+
+other_states_deaths = deaths_copy[
+    (deaths_copy["State"] != "Florida")
+    & (
+        (
+            deaths_copy["Drug/Alcohol Induced Cause"]
+            == "Drug poisonings (overdose) Unintentional (X40-X44)"
+        )
+        | (
+            deaths_copy["Drug/Alcohol Induced Cause"]
+            == "Drug poisonings (overdose) Undetermined (Y10-Y14)"
+        )
+        | (
+            deaths_copy["Drug/Alcohol Induced Cause"]
+            == "Drug poisonings (overdose) Suicide (X60-X64)"
+        )
+    )
+]
```

Comment:
Hey Annapurna, very good subset of different levels of the category. You might dont need this if you change to import the newest dataset uploaded by Alex.